File: cs/Problems/Day5Test.cs
namespace aoc24.Problems;

public class Day5Test
{
    private readonly Day5 sut = new();

    [Theory, InlineData(143)]
    public void TestSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day5_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(7307)]
    public void FullSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day5_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day5_test.go
package problems

import "testing"

func TestDay5_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day5{}, "day5_1", 143)
}

func TestDay5_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day5{}, "day5_2", 7307)
}
